src/storybook/story-test-runner/open-story/hermione-open-story.ts

```diff
@@ -0,0 +1,159 @@
+        const channel = (window as StorybookWindow).__STORYBOOK_ADDONS_CHANNEL__;
+        const result: StoryLoadResult = { rootSelector: "" };
+
+        function doneJson(value: StoryLoadResult): void {
+            channel.off("playFunctionThrewException", onPlayFunctionThrewException);
+            channel.off("storyRendered", onStoryRendered);
+            channel.off("storyMissing", onStoryMissing);
+            channel.off("storyThrewException", onStoryThrewException);
+            channel.off("storyErrored", onStoryErrored);
+
+            done(JSON.stringify(value));
+        }
+
+        function onPlayFunctionThrewException(): void {
+            result.playFunctionError = "an unknown exception was thrown";
+        }
+
+        function onStoryRendered(): void {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            result.rootSelector = ["#storybook-root", "#root"].find(function (selector) {
+                return document.querySelector(selector);
+            })!;
+
+            if (result.playFunctionError) {
+                const instrumenter = (window as StorybookWindow).__STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__;
+                const storyState = instrumenter ? instrumenter[storyId] : null;
+                const storyLastCall =
+                    storyState && storyState.calls ? storyState.calls[storyState.calls.length - 1] : null;
+
+                if (storyLastCall && storyLastCall.exception) {
+                    result.playFunctionError = storyLastCall.exception.message;
+                }
+            }
+
+            doneJson(result);
+        }
+
+        function onStoryMissing(): void {
+            const errorMessage = (document.querySelector("#error-stack") as unknown as HTMLElement).innerText;
+            const regExpMatch = /\w+:\s+x\s+/.exec(errorMessage);
+
+            if (regExpMatch) {
+                result.loadError = errorMessage.slice(regExpMatch[0].length);
+            } else {
+                result.loadError = errorMessage;
+            }
+
+            doneJson(result);
+        }
+
+        function onStoryThrewException(exception: Error): void {
+            if (exception.message && !exception.message.startsWith("ignoredException")) {
+                result.loadError = exception.message;
+
+                doneJson(result);
+            }
+        }
+
+        function onStoryErrored(error: { title: string }): void {
+            result.loadError = error.title;
+
+            doneJson(result);
+        }
+
+        channel.once("playFunctionThrewException", onPlayFunctionThrewException);
+        channel.once("storyRendered", onStoryRendered);
+        channel.once("storyMissing", onStoryMissing);
+        channel.once("storyThrewException", onStoryThrewException);
+        channel.once("storyErrored", onStoryErrored);
+
+        if (remountOnly) {
+            channel.emit("forceRemount", { storyId });
+        } else {
```

Comment:
We don't need to switch to another story and back, if we are on the right story at the moment.
We can just `forceRemount`